### DISCUSSION TITLE
Feat: zoxide integration

### DISCUSSION THREAD

**Discussion by rwedmonds** [aws/codewhisperer-command-line-discussions#82] [Feature request]
### Checks

- [X] I could not find an existing setting that solved my issue
- [X] I searched [the discussions in this repo](https://github.com/aws/codewhisperer-command-line-discussions/discussions?q=) and there are no duplicates of my feature request

### Description:
> It would be great if this worked with zoxide, perhaps taking into account its ranking system to display suggestions in the same manner as it does with the `cd` command.

**Comment by brendanfalk** [comment]
Thanks for reporting! We will investigate, however, not likely on our top priority list for a little while

**Comment by TizzySaurus** [comment]
@rwedmonds

CodeWhisperer CLI already has `zoxide` integration, presumably copied from the [Fig spec for zoxide](https://github.com/withfig/autocomplete/blob/master/src/zoxide.ts):
<img width="572" alt="image" src="https://github.com/aws/codewhisperer-command-line-discussions/assets/47674925/f098c3b2-3ad0-4cf9-83a5-a56a13578157">

It also sort of works with `z`, just will only display directories in your `z` history, which means if you want to go to a directory that you've not yet visited with `z`, you don't get any autocomplete (maybe it should fall back to `cd` logic so that you still get path completion). [Fig spec for z](https://github.com/withfig/autocomplete/blob/master/src/z.ts). It even ranks based on `z`'s ordering (see the 'score' at bottom): <img width="579" alt="image" src="https://github.com/aws/codewhisperer-command-line-discussions/assets/47674925/fbe52ac4-5858-4b3f-a8f6-fcebaacb60de">

What exactly is missing that you're after @rwedmonds?

**Reply by rwedmonds** [reply]
Weird. The second one was actually what I was after, but it doesn't work for me. The `cd` command works, `z` does not.

![cd_works](https://github.com/aws/codewhisperer-command-line-discussions/assets/105442447/dc92884d-0973-4851-8cb1-fada50e758e6)

![z_does_not](https://github.com/aws/codewhisperer-command-line-discussions/assets/105442447/af87bea1-79b4-4f28-a1e7-94e83caebd44)

**Reply by ScottRobinson03** [reply]
> The second one was actually what I was after, but it doesn't work for me. 

@rwedmonds As per the above comment:
> It also sort of works with z, just will only display directories in your z history, which means if you want to go to a directory that you've not yet visited with z, you don't get any autocomplete (maybe it should fall back to cd logic so that you still get path completion).

The integration with `z` only gives autocomplete for paths that you've already visited with `z`. That means that the first time you go to a path with `z`, you don't get autocomplete. Once you've visited it, you get autocomplete. In this case, manually type `z .config`, then if you `cd ..` and type `z ` you should see ".config" (or rather, `/User/redmonds/.config`) come up as a suggestion. 

I've created a bug report for tracking purposes in #114.

**Comment by grant0417** [comment]
If you want a better integration with zoxide all of our completions are opensource and PRs are welcome: https://github.com/withfig/autocomplete/blob/master/src/z.ts
